### PR TITLE
add templateName to template to contains template name

### PR DESCRIPTION
**Pull request by yuzp1996**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [`spec` PR link](https://github.com/katanomi/spec) included
- [ ] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [ ] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [ ] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [ ] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

### CODE REVIEW

File: apis/meta/v1alpha1/metadata_type.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+*/
+
+package v1alpha1
+
+import "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+
+// Metadata contains name labels and annotations
+type Metadata struct {
+	// Name represent name
+	Name string `json:"name"`
```
Comment (l-qing) [inline apis/meta/v1alpha1/metadata_type.go:24]: 这个是可选字段？

File: apis/meta/v1alpha1/metadata_type.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	// PipelineTaskMetadata contains labels and annotations
+	v1beta1.PipelineTaskMetadata
```
Comment (l-qing) [inline apis/meta/v1alpha1/metadata_type.go:26]: 一般 inline 的结构体是放在最前面，且与命名字段有空行？

File: apis/meta/v1alpha1/metadata_type.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+package v1alpha1
+
+import "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+
+// Metadata contains name labels and annotations
+type Metadata struct {
+	// PipelineTaskMetadata contains labels and annotations
+	// +optional
+	v1beta1.PipelineTaskMetadata `json:",inline,omitempty"`
```
Comment (l-qing) [inline apis/meta/v1alpha1/metadata_type.go:25]: inline 的好像没有 omitempty 之说？

**Review by l-qing — APPROVED** [review]

File: apis/meta/v1alpha1/metadata_type.go (hunk 3 above)
Comment (yuzp1996) [reply apis/meta/v1alpha1/metadata_type.go]: 企业微信截图_fa736e95-95b3-4c1f-bb1b-9df30d372099

tekton 中存在类似的使用  不过为了和我们自己的使用方式对齐  这里会去掉

**Pull request merged**